File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum=nums[0]
        total=0
        for i in range(0,len(nums)):
            total+=nums[i]
            max_sum=max(total,max_sum)
            if total<0:
                total=0
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.java
import java.util.*;
class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum=Integer.MIN_VALUE;
        int add=0;
        for (int i=0;i<nums.length;i++){
            add+=nums[i];
            if (add>max_sum){
                max_sum=add;
            }
            if (add<0){
                add=0;
            }
        }
        return max_sum;
    }
}